plugin/servergen.go

```diff
@@ -45,61 +44,217 @@ func (p *OrmPlugin) generateDefaultServer(file *generator.FileDescriptor) {
 func (p *OrmPlugin) generateCreateServerMethod(service *descriptor.ServiceDescriptorProto, method *descriptor.MethodDescriptorProto) {
 	inType, outType, methodName, svcName := p.getMethodProps(service, method)
 	p.generateMethodSignature(inType, outType, methodName, svcName)
-	p.P(`var out `, p.TypeName(outType))
-	p.P(`res, err := DefaultCreate`, strings.TrimPrefix(methodName, "Create"), `(ctx, in.GetPayload(), m.DB)`)
-	p.P(`out.Result = res`)
-	p.P(`return &out, err`)
-	p.P(`}`)
+	follows, typeName := p.followsCreateConventions(inType, outType)
+	if follows {
+		p.P(`res, err := DefaultCreate`, typeName, `(ctx, in.GetPayload(), m.DB)`)
+		p.P(`if err != nil {`)
+		p.P(`return nil, err`)
+		p.P(`}`)
+		p.P(`return &`, p.TypeName(outType), `{Result: res}, nil`)
+		p.P(`}`)
+	} else {
+		p.generateEmptyBody(outType)
+	}
+}
+
+func (p *OrmPlugin) followsCreateConventions(inType generator.Object, outType generator.Object) (bool, string) {
+	inMsg := inType.(*generator.Descriptor)
+	outMsg := outType.(*generator.Descriptor)
+	var inTypeName string
+	var typeOrmable bool
+	for _, field := range inMsg.Field {
+		if field.GetName() == "payload" {
+			gType, _ := p.GoType(inMsg, field)
+			inTypeName = strings.TrimPrefix(gType, "*")
+			if _, exists := convertibleTypes[inTypeName]; exists {
+				typeOrmable = true
+			}
+		}
+	}
+	var outTypeName string
+	for _, field := range outMsg.Field {
+		if field.GetName() == "result" {
+			gType, _ := p.GoType(outMsg, field)
+			outTypeName = strings.TrimPrefix(gType, "*")
+		}
+	}
+	if inTypeName == outTypeName && typeOrmable {
+		return true, lintName(inTypeName)
+	}
+	return false, ""
 }
 
 func (p *OrmPlugin) generateReadServerMethod(service *descriptor.ServiceDescriptorProto, method *descriptor.MethodDescriptorProto) {
 	inType, outType, methodName, svcName := p.getMethodProps(service, method)
 	p.generateMethodSignature(inType, outType, methodName, svcName)
-	p.P(`var out `, p.TypeName(outType))
-	p.P(`res, err := DefaultRead`, strings.TrimPrefix(methodName, "Read"), `(ctx, in.GetPayload(), m.DB)`)
-	p.P(`out.Result = res`)
-	p.P(`return &out, err`)
-	p.P(`}`)
+	follows, typeName := p.followsReadConventions(inType, outType)
+	if follows {
+		p.P(`res, err := DefaultRead`, typeName, `(ctx, &`, typeName, `{Id: in.GetId()}, m.DB)`)
```

Comment:
The choice to use golint compliant names for the ORMable objects is causing way more trouble than I anticipated.
In these statements, the first typeName is linted (since I thought of the methods as CURDLing an ORM object) and the second one is not linted, since it's coming from the pb generated file. This will only make a difference on names with initialisms though (ACL, ID, API, etc.) so it's a horrible 'gotcha'.